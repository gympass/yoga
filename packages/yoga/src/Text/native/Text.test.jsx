import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from '../..';
import { Text } from '.';

describe('<Text />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with Text', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text.H1>Live the mission</Text.H1>
          <Text.H2>Live the mission</Text.H2>
          <Text.H3>Live the mission</Text.H3>
          <Text.H4>Live the mission</Text.H4>
          <Text>Live the mission</Text>
          <Text.Small>Live the mission</Text.Small>
          <Text.Tiny>Live the mission</Text.Tiny>
          <Text.Regular>Live the mission</Text.Regular>
          <Text.Medium>Live the mission</Text.Medium>
          <Text.Bold>Live the mission</Text.Bold>
          <Text.Black>Live the mission</Text.Black>
          <Text.SectionTitle>Live the mission</Text.SectionTitle>
          <Text.SmallestException>Live the mission</Text.SmallestException>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with Text variant', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text.H1 variant="primary">Live the mission</Text.H1>
          <Text.H2 variant="secondary">Live the mission</Text.H2>
          <Text.H3 variant="tertiary">Live the mission</Text.H3>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with Text light', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text.H1 light>Live the mission</Text.H1>
          <Text.H2 light>Live the mission</Text.H2>
          <Text.H3 light>Live the mission</Text.H3>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with Text size', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text.Small size="xsmall">Live the small mission</Text.Small>
          <Text.Tiny size="medium">Live the medium mission</Text.Tiny>
          <Text.Regular size="xlarge">Live the xlarge mission</Text.Regular>
          <Text.Medium size="xxxlarge">Live the xxxlarge mission</Text.Medium>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with inverted Text', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text inverted>Live the mission</Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with truncated lines Text', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text numberOfLines={1}>
            Live the mission
            <br />
            even in one line
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with border prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text border="small" bRadius="small">
            Live the mission
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with background color prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text backgroundColor="vibin">Live the mission</Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with elevation prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text elevation="medium">Live the mission</Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with spacing prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text m="medium" paddingVertical="large" paddingHorizontal="small">
            Live the mission
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with typography prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text
            color="text.primary"
            fontSize="small"
            fontWeight="medium"
            lineHeight="small"
          >
            Live the mission
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with display prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text display="flex">Live the mission</Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with position prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text position="absolute" top="small" left="small">
            Live the mission
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with flexes prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text display="flex" flexGrow={1}>
            Live the mission
          </Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with overflows prop system', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Text overflow="scroll">Live the mission</Text>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
