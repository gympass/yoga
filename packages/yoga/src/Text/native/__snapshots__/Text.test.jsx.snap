// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Snapshots should match snapshot with Text 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 48,
          "fontWeight": "500",
          "lineHeight": 56,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 40,
          "fontWeight": "500",
          "lineHeight": 48,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 32,
          "fontWeight": "500",
          "lineHeight": 40,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 24,
          "fontWeight": "500",
          "lineHeight": 32,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    inverted={false}
    light={false}
    size="medium"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 16,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "400",
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "500",
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "700",
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "900",
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 1,
          "lineHeight": 16,
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 10,
          "fontWeight": "400",
          "lineHeight": 12,
        },
      ]
    }
  >
    Live the mission
  </Text>,
]
`;

exports[`<Text /> Snapshots should match snapshot with Text light 1`] = `
Array [
  <Text
    light={true}
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 48,
          "fontWeight": "300",
          "lineHeight": 56,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    light={true}
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 40,
          "fontWeight": "300",
          "lineHeight": 48,
        },
      ]
    }
  >
    Live the mission
  </Text>,
  <Text
    light={true}
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 32,
          "fontWeight": "300",
          "lineHeight": 40,
        },
      ]
    }
  >
    Live the mission
  </Text>,
]
`;

exports[`<Text /> Snapshots should match snapshot with Text size 1`] = `
Array [
  <Text
    size="xsmall"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 20,
        },
      ]
    }
  >
    Live the small mission
  </Text>,
  <Text
    size="medium"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 12,
          "fontWeight": "400",
          "lineHeight": 16,
        },
      ]
    }
  >
    Live the medium mission
  </Text>,
  <Text
    size="xlarge"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 24,
          "fontWeight": "400",
        },
      ]
    }
  >
    Live the xlarge mission
  </Text>,
  <Text
    size="xxxlarge"
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 40,
          "fontWeight": "500",
        },
      ]
    }
  >
    Live the xxxlarge mission
  </Text>,
]
`;

exports[`<Text /> Snapshots should match snapshot with Text variant 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#D8385E",
          "fontFamily": "Rubik",
          "fontSize": 48,
          "fontWeight": "500",
          "lineHeight": 56,
        },
      ]
    }
    variant="primary"
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#231B22",
          "fontFamily": "Rubik",
          "fontSize": 40,
          "fontWeight": "500",
          "lineHeight": 48,
        },
      ]
    }
    variant="secondary"
  >
    Live the mission
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "Rubik",
          "fontSize": 32,
          "fontWeight": "500",
          "lineHeight": 40,
        },
      ]
    }
    variant="tertiary"
  >
    Live the mission
  </Text>,
]
`;

exports[`<Text /> Snapshots should match snapshot with background color prop system 1`] = `
<Text
  backgroundColor="vibin"
  inverted={false}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "backgroundColor": "#D8385E",
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with border prop system 1`] = `
<Text
  bRadius="small"
  border="small"
  inverted={false}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "borderColor": "black",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with display prop system 1`] = `
<Text
  display="flex"
  inverted={false}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "display": "flex",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with elevation prop system 1`] = `
<Text
  elevation="medium"
  inverted={false}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
        "shadowColor": "rgba(152, 152, 166, 0.2)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 5,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with flexes prop system 1`] = `
<Text
  display="flex"
  flexGrow={1}
  inverted={false}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "display": "flex",
        "flexGrow": 1,
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with inverted Text 1`] = `
<Text
  inverted={true}
  light={false}
  size="medium"
  style={
    Array [
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with overflows prop system 1`] = `
<Text
  inverted={false}
  light={false}
  overflow="scroll"
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
        "overflow": "scroll",
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with position prop system 1`] = `
<Text
  inverted={false}
  left="small"
  light={false}
  position="absolute"
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "left": 16,
        "lineHeight": 24,
        "position": "absolute",
        "top": 16,
      },
    ]
  }
  top="small"
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with spacing prop system 1`] = `
<Text
  inverted={false}
  light={false}
  m="medium"
  paddingHorizontal="small"
  paddingVertical="large"
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
        "marginBottom": 20,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 20,
        "paddingBottom": 24,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
      },
    ]
  }
>
  Live the mission
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with truncated lines Text 1`] = `
<Text
  inverted={false}
  light={false}
  numberOfLines={1}
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
    ]
  }
>
  Live the mission
  <br />
  even in one line
</Text>
`;

exports[`<Text /> Snapshots should match snapshot with typography prop system 1`] = `
<Text
  color="text.primary"
  fontSize="small"
  fontWeight="medium"
  inverted={false}
  light={false}
  lineHeight="small"
  size="medium"
  style={
    Array [
      Object {
        "color": "#231B22",
        "fontFamily": "Rubik",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 20,
      },
    ]
  }
>
  Live the mission
</Text>
`;
